src/main/java/org/folio/circulation/infrastructure/storage/inventory/ItemRepository.java

```diff
@@ -254,6 +257,20 @@ private CompletableFuture<Result<MultipleRecords<Item>>> fetchItems(Collection<S
     final var mapper = new ItemMapper();
 
     return finder.findByIds(itemIds)
+      .thenApply(mapResult(identityMap::add))
+      .thenApply(mapResult(records -> records.mapRecords(mapper::toDomain)))
+      .thenCompose(res -> res.value().getTotalRecords() == itemIds.size()
```

Comment:
What will happen if we're trying to find 10 items by ID but one of them was deleted and we only received 9 items in response? It looks like we're going to try to find the 10th item as DCB item. Was that the intention, to make it possible to pass a mix of DCB and non-DCB item IDs in this method and find all of them?

Reply:
Yes correct , the intention is to find all dcb and non-dcb items, And its correct because we are not sure in above case whether the item is deleted from inventory or its a DCB item. thanks